client/sites/overview/components/test/hosting-overview.tsx

```diff
@@ -0,0 +1,318 @@
+		<QueryClientProvider client={ queryClient }>
+			<Provider store={ store }>{ element }</Provider>
+		</QueryClientProvider>
+	);
+};
+
+const mockSiteId = 123;
+
+type MockSiteOptions = {
+	is_wpcom_atomic?: boolean;
+	migrationStatus?: MigrationStatus;
+	migrationSticker?: string;
+};
+
+const buildMockSite = ( {
+	is_wpcom_atomic = false,
+	migrationStatus = MigrationStatus.UNKNOWN,
+	migrationSticker,
+}: MockSiteOptions ): SiteDetails => {
+	return {
+		ID: mockSiteId,
+		slug: 'test-site.wordpress.com',
+		URL: 'https://test-site.wordpress.com',
+		capabilities: {
+			activate_plugins: false,
+			activate_wordads: false,
+			delete_others_posts: true,
+			delete_posts: true,
+			delete_users: true,
+			edit_others_pages: true,
+			edit_others_posts: true,
+			edit_pages: true,
+			edit_posts: true,
+			edit_theme_options: true,
+			edit_users: true,
+			list_users: true,
+			manage_categories: true,
+			manage_options: true,
+			moderate_comments: true,
+			own_site: true,
+			promote_users: true,
+			publish_posts: true,
+			remove_users: true,
+			upload_files: true,
+			update_plugins: false,
+			view_hosting: true,
+			view_stats: true,
+		},
+		description: 'Test site description',
+		domain: 'test-site.wordpress.com',
+		jetpack: true,
+		launch_status: 'launched',
+		locale: 'en_US',
+		is_wpcom_atomic,
+		logo: {
+			id: '123',
+			sizes: [ 'thumbnail', 'medium', 'large' ],
+			url: 'https://test-site.wordpress.com/logo.png',
+		},
+		name: 'Test Site',
+		title: 'Test Site',
+		site_migration: {
+			status: migrationStatus,
+			last_modified: '2025-03-03',
+			migration_status: migrationSticker,
+		},
+	};
+};
+
+const mockCalypsoState = {
+	currentUser: {
+		capabilities: {},
+		flags: [],
+	},
+	purchases: {
+		isFetchingSitePurchases: true,
+	},
+	sites: {
+		items: {
+			[ mockSiteId ]: buildMockSite( {} ),
+		},
+		plans: {
+			[ mockSiteId ]: {
+				data: null,
+				error: null,
+				hasLoadedFromServer: false,
+				isRequesting: true,
+			},
+		},
+		requesting: {},
+		requestingAll: false,
+	},
+	ui: {
+		selectedSiteId: mockSiteId,
+	},
+};
+
+const mockRequestSiteAction = {
+	type: 'SITE_REQUEST',
+	siteId: mockSiteId,
+};
+
+describe( 'HostingOverview', () => {
+	beforeAll( () => {
+		Object.defineProperty( window, 'location', {
+			value: { ...originalLocation, assign: jest.fn() },
+		} );
+	} );
+
+	afterAll( () => {
+		Object.defineProperty( window, 'location', originalLocation );
+	} );
+
+	beforeEach( () => {
+		jest.resetAllMocks();
+	} );
+
+	describe( 'when refresh parameter is not true', () => {
+		beforeEach( () => {
+			window.location.search = '';
+		} );
+
+		it( 'should load the page and not request site details', () => {
+			requestSiteSpy = jest.spyOn( siteActions, 'requestSite' );
+
+			const { getByTestId, getByText } = localRenderWithProvider( <HostingOverview />, {
+				initialState: mockCalypsoState,
+			} );
+
+			expect( requestSiteSpy ).not.toHaveBeenCalled();
+
+			waitFor( () => {
```

Comment:
```suggestion
			async waitFor( () => {
```

Reply:
waitFor requires an await to return errors.

Reply:
😬 🤦  Addressed -- this was hiding some far more subtle issues with the state/redux handling!